### PR TITLE
fix(python): add support for cli args

### PR DESCRIPTION
**Pull request by gautamkrishnar**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:
Added command line args support to ipython alias, fixes: #12153

## Other comments:
n/a

### CODE REVIEW

File: plugins/python/python.plugin.zsh [hunk 1]
```diff
@@ -44,7 +44,7 @@ function pyuserpaths() {
 alias pygrep='grep -nr --include="*.py"'
 
 # Run proper IPython regarding current virtualenv (if any)
-alias ipython="python3 -c 'import IPython; IPython.terminal.ipapp.launch_new_instance()'"
+alias ipython='python3 -c "import IPython, sys; IPython.start_ipython(argv=sys.argv[1:])"'
```
Comment (carlosala) [inline plugins/python/python.plugin.zsh:47]: It seems that it's not needed to pass `argv`, it should already parse them: https://ipython.readthedocs.io/en/stable/api/generated/IPython.html#IPython.start_ipython

**Comment by carlosala** [comment]
Even more, this is the script that launches `ipython` in my system (`v8.2.0`).
```python
#!/usr/bin/python
# -*- coding: utf-8 -*-
import re
import sys
from IPython import start_ipython
if __name__ == "__main__":
    sys.argv[0] = re.sub(r"(-script\.pyw|\.exe)?$", "", sys.argv[0])
    sys.exit(start_ipython())
```
We could simplify it to:
```zsh
alias ipython='python3 -c "import IPython, sys; sys.exit(IPython.start_ipython())"'
```

File: plugins/python/python.plugin.zsh (hunk 1 above)
Comment (gautamkrishnar) [reply plugins/python/python.plugin.zsh:47]: Interesting 😄

**Comment by gautamkrishnar** [comment]
@carlosala updated the alias to your suggestion. Please squash the pr before merging.

**Review by carlosala — APPROVED** [review]
LGTM!

**Comment by carlosala** [comment]
Thanks for your contribution!

**Comment by gautamkrishnar** [comment]
Thanks for reviewing.

**Pull request merged**